srecord/arglex.cc

```diff
@@ -420,7 +420,7 @@ deprecated_warning(const char *deprecated_name, const char *preferred_name)
 //
 
 int
-srecord::arglex::token_next(void)
```

Comment:
I'm used to `(void)` myself, because I also do C.
But the C++ Core Guideline says [this](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-void): *Even Dennis Ritchie deemed void f(void) an abomination*

Reply:
Who am I to argue with Dennis Ritchie!. Definitely pull this in.